lib/bundler/plugin_dsl.rb

```diff
@@ -0,0 +1,49 @@
+require 'bundler/dsl'
+module Bundler
+  class PluginDsl
+
+    def self.evaluate(gemfile, lockfile, unlock)
+      builder = new
+      builder.eval_gemfile_for_plugins(gemfile)
+      builder.to_plugin_definition(lockfile, unlock)
+    end
+
+    def initialize
+      @dsl = Dsl.new
+      methods = Dsl.instance_methods(false)
+      methods.delete(:plugin)
+
+      self.class.instance_eval do
```

Comment:
why is this done for the class each `initialize`? Should be done at the class scope